### ISSUE TITLE
Unicode error: invalid utf8 when using special characters from Polish alphabet

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#28445]
**Godot version:**
3.2.dev.custom_build. 260530f

**OS/device including version:**
Linux Mint 19

**Issue description:**
When I set name of project "Zażółć gęślą jaźń" (default text containing all special Polish letters to check support in fonts), then Godot shows error:
```
Unicode error: invalid utf8
```
This is strange because all this characters are present in Unicode  https://en.wikipedia.org/wiki/Polish_alphabet#Computer_encoding

Also when I type as name of project just "ó", then this error is shown:
```
Unicode error: no space left
```

EDIT:
Also "Remove Missing Project" in Project Manager is triggered by this bug, because it think, that project doesn't exists and remove it

**Comment by gdupr** [comment]
I tested with your phrase and with the "ó"(00f3) char.

It seems that saving a project called "Zażółć gęślą jaźń" works fine for me. Saving(Save & Quit) a project called "ó" triggers a crash. The same with "ź" and probably the other characters also.

I also tested some mandarin and japanese characters and it works with no issues. But there's a quick hack to fix this:
```
$ git diff
diff --git a/core/ustring.cpp b/core/ustring.cpp
index 78feddb22..52f34eb48 100644
--- a/core/ustring.cpp
+++ b/core/ustring.cpp
@@ -1416,7 +1416,7 @@ bool String::parse_utf8(const char *p_utf8, int p_len) {
                                uint8_t c = *ptrtmp;

                                /* Determine the number of characters in sequence */
-                               if ((c & 0x80) == 0)
+                               if ((c & 0xC0) == 0x80 || (c & 0x80) == 0)
                                        skip = 0;
                                else if ((c & 0xE0) == 0xC0)
                                        skip = 1;
@@ -1470,7 +1470,7 @@ bool String::parse_utf8(const char *p_utf8, int p_len) {
                int len = 0;

                /* Determine the number of characters in sequence */
-               if ((*p_utf8 & 0x80) == 0)
+               if ((*p_utf8 & 0xC0) == 0x80 || (*p_utf8 & 0x80) == 0)
                        len = 1;
                else if ((*p_utf8 & 0xE0) == 0xC0)
                        len = 2;

```
Got it working with your characters..

**Comment by qarmin** [comment]
Also when I set name project to aąąąśśśśśśś , then Godot recognize it as a[[[[[[[

**Comment by timothyqiu** [comment]
This may have been fixed by #34472. I can't reproduce on the current master.

**Comment by qarmin** [comment]
I can't reproduce too.
